packages/sign-client/src/controllers/engine.ts

```diff
@@ -1107,6 +1231,13 @@ export class Engine extends IEngine {
   };
 
   private deleteProposal: EnginePrivate["deleteProposal"] = async (id, expirerHasDeleted) => {
+    if (expirerHasDeleted) {
+      try {
+        const proposal = this.client.proposal.get(id);
+        const event = this.client.core.eventClient.getEvent({ topic: proposal.pairingTopic });
+        event?.setError(EVENT_CLIENT_SESSION_ERRORS.proposal_expired);
+      } catch (error) {}
```

Comment:
Curious where could this throw ? Should we just add a simple log to eventually catch weird cases ?

Reply:
`const proposal = this.client.proposal.get(id);`
the storage throws if the `id` is not found. Imo logging is not nessesary because its not actionable by users and doesn't interrupt the flow. The method tries to delete a proposal that is already deleted so no issues there